### PR TITLE
Add ConfigJavaInfo to jar jar aspect

### PR DESCRIPTION
**Pull request by ramyanaga**
Problem:
We need certain transitive runtime jars to be added to the current jars list.

Solution:
This PR creates a provider ConfigJavaInfo that a target can expose with specific jars that need to be added to current jars

### CODE REVIEW

**Review by eed3si9n — APPROVED** [review]

File: jar_jar_aspect.bzl [hunk 1]
```diff
@@ -57,10 +61,15 @@ def _jar_jar_aspect_impl(target, ctx):
...
     current_jars = [j for j in target[JavaInfo].runtime_output_jars]
+
     # For some outputs, like those from the built in java proto aspect, the runtime output jars is empty, but the jars
     # exist instead in the java_outputs section.
     current_jars.extend([e.class_jar for e in target[JavaInfo].java_outputs])
+    java_info_runtime_deps = []
 
+    if ConfigJavaInfo in target:
+        java_info_runtime_deps.append(target[ConfigJavaInfo].config_java_info)
+        #current_jars.extend([e.class_jar for e in target[ConfigJavaInfo].config_java_info.java_outputs])
```
Comment (johnynek) [inline jar_jar_aspect.bzl:72]: we some remove this commented code.

File: jar_jar_aspect.bzl [hunk 2]
```diff
@@ -6,6 +6,10 @@ ShadedJars = provider(fields = [
     "transitive_shaded",
 ])
 
+ConfigJavaInfo = provider(fields = [
```
Comment (johnynek) [inline jar_jar_aspect.bzl:9]: I find this name a bit confusing. I would rather name it something related to `runtime_deps`. But also, I'm confused why we need this... seems like we shouldn't...

**Comment by johnynek** [comment]
You said: "We need certain transitive runtime jars to be added to the current jars list." Can you be more explicit?

Why do we need certain transitive runtime jars added? Is it because the aspect isn't recursing properly? Maybe that's it because the "certain jars" are coming from a rule that the aspect can't traverse? Maybe we should fix our internal rule...

File: jar_jar_aspect.bzl (hunk 2 above)
Comment (eed3si9n) [reply jar_jar_aspect.bzl:9]: The problem is that this rule makes the assumption that the target only generates one output JAR, and if it ever has a runtime dependency the target would have an attributed called `runtime_deps`:

```python
    if hasattr(ctx.rule.attr, "runtime_deps"):
        for d in ctx.rule.attr.runtime_deps:
            if ShadedJars in d:
                shaded_jars = d[ShadedJars]
                transitive_shaded.append(depset([shaded_jars]))
                transitive_shaded.append(shaded_jars.transitive_shaded)
                java_info_runtime_deps.append(shaded_jars.java_info)
```

The target we have generates two output one normal JAR and the second JAR for JSON, and as far as we can tell it won't get included into the resulting `JavaInfo` from this aspect.

**Pull request merged**